src/ResourceInterface.php

```diff
@@ -11,61 +11,51 @@
 
 namespace Tobscure\JsonApi;
 
-interface SerializerInterface
+interface ResourceInterface
 {
     /**
-     * Get the type.
-     *
-     * @param mixed $model
+     * Get the resource type.
      *
      * @return string
      */
-    public function getType($model);
+    public function getType();
 
     /**
-     * Get the id.
-     *
-     * @param mixed $model
+     * Get the resource ID.
      *
      * @return string
      */
-    public function getId($model);
+    public function getId();
 
     /**
-     * Get the attributes array.
+     * Get the resource attributes.
      *
-     * @param mixed $model
      * @param array|null $fields
      *
      * @return array
      */
-    public function getAttributes($model, array $fields = null);
+    public function getAttributes(array $fields = null);
```

Comment:
Hmm I think that probably creates more pain than it saves when you do want to implement it, because you then have two functions that are calculating attributes:

```php
public function getAttributes() {
    return [
        'foo' => $this->calculateExpensiveAttribute()
    ];
}
public function getSparseAttributes(array $fields) {
    $attributes = [];

    if (in_array('foo', $fields)) {
        $attributes['foo'] = $this->calculateExpensiveAttribute();
    }

    return $attributes;
}
```

vs. what we have now:

```php
public function getAttributes(array $fields = null) {
    $attributes = [];

    if ($fields === null || in_array('foo', $fields)) {
        $attributes['foo'] = $this->calculateExpensiveAttribute();
    }

    return $attributes;
}
```

For implementations that don't care about sparse fieldsets, I'm pretty sure you can leave out the $fields argument completely (as in the `getAttributes()` method of my first example) since it has a default of null?